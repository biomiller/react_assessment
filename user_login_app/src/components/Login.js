import React, { Component } from 'react';
import axios from "axios";

import {
    Button
} from 'reactstrap';


export class Login extends Component {

    constructor() {
        super();
        this.state = {
            password: "",
            username: "",
            login:""
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();

        let username = e.target[0].value
        let password = e.target[1].value

        axios
            .get(`http://localhost:5000/user/name/${username}/${password}`)
            .then(response => {
                if (response.data.Status === "Not Logged In") {
                    this.setState({
                        username: "",
                        password: "Incorrect password."
                    });
                } else if (response.data.Status === "Logged In"){
                    this.props.userLoggedIn(username);
                    this.setState({
                        username: "",
                        password: "",
                        login: "Successfully logged in."
                    });
                }
            })
            .catch(err => {
                console.log(err);
                this.setState({
                    username: "User not found.",
                    password: ""
                })
            })
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>

                <h1>Login</h1>

                <input required type="text" placeholder="Username"></input>
                <p>{this.state.username}</p>

                <input required type="password" placeholder="Password"></input>
                <p>{this.state.password}</p>

                <Button type="submit">Submit</Button>
                <br></br>
                <p>{this.state.login}</p>


            </form>

        );
    }
}