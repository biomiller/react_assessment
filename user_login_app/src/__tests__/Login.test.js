import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import { Login } from "../components/Login.js";


it('is Login rendered', () => {
    const component = renderer
    .create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    .toJSON();

    expect(component).toMatchSnapshot();
});



