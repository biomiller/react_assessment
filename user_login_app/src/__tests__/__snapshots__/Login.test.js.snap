// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is Login rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Login
  </h1>
  <input
    placeholder="Username"
    required={true}
    type="text"
  />
  <p>
    
  </p>
  <input
    placeholder="Password"
    required={true}
    type="password"
  />
  <p>
    
  </p>
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <br />
  <p>
    
  </p>
</form>
`;
